File: trunk/staff/tools/codegen/template/jsclient/Serialization.js
#foreach $(Interface.Enums)
#ifeq($(Enum.Extern),0) // do not serialize extern type
#cginclude "EnumSerialization.js"
#ifeqend
#end
#foreach $(Interface.Structs)
#cginclude "StructSerialization.js"
#end
#foreach $(Interface.Typedefs)
#ifeq($(Typedef.Extern),0) // do not serialize extern type

// typedef $(Typedef.NsName.!dot)
var o$(Typedef.NsName.!mangle) = staff.object("$(Typedef.NsName.!dot)");
o$(Typedef.NsName.!mangle).serialize = function(oElement, oType)
{
#var Type oType
#var Element oElement
#context $(Typedef.DataType)
#cginclude "TypeSerialize.js"
#contextend
  return oElement;
}

o$(Typedef.NsName.!mangle).deserialize = function(oElement)
{
  var oType = \
#var Type oType
#var Element oElement
#context $(Typedef.DataType)
#cginclude "TypeDeserialize.js"
#contextend
  return oType;
}

#ifeqend //ifeq($(Typedef.Extern),0) // do not serialize extern type
#end // foreach $(Interface.Structs)

File: tags/staff-2.0.0-b1/staff/tools/codegen/templates/common/Serialization.cpp
namespace staff
{
#ifneq($(Interface.Enums.$Count),0)
///////////////////////////////////////////////////////////////////////////////////////////////////////
// enums
#foreach $(Interface.Enums)
#ifeq($(Enum.Extern),false) // do not serialize extern type
#cginclude "EnumSerialization.cpp"
#ifeqend
#end

#ifeqend // enums
#ifneq($(Interface.Structs.$Count),0)
///////////////////////////////////////////////////////////////////////////////////////////////////////
// struct serializators/deserializators
#foreach $(Interface.Structs)
#cginclude "StructSerialization.cpp"
#end

#ifeqend
#ifneq($(Interface.Typedefs.$Count),0)
///////////////////////////////////////////////////////////////////////////////////////////////////////
// typedef serializators
#foreach $(Interface.Typedefs)
#ifeq($(Typedef.Extern),false) // do not serialize extern type

DataObject& SerializeTypedef_$(Typedef.NsName.!mangle)(DataObject& rdoParam, const $(Typedef.NsName)& rtType)
{
#cgpushvars
#var bUseParentElement $(Typedef.Options.*useParentElement)
#var bDontWriteNs true
#context $(Typedef.DataType)
#var sParamName
#var sParam rtType
#var sdoParam rdoParam
#cginclude "TypeSerialization.cpp"
#contextend
#cgpopvars
  return rdoParam;
}

#ifeqend //ifeq($(Typedef.Extern),false) // do not serialize extern type
#end // foreach $(Interface.Structs)
#ifeqend

///////////////////////////////////////////////////////////////////////////////////////////////////////
// typedef deserializators
#foreach $(Interface.Typedefs)
#ifeq($(Typedef.Extern),false)
const DataObject& DeserializeTypedef_$(Typedef.NsName.!mangle)\
(const DataObject& rdoParam, $(Typedef.NsName)& rtType)
{
#cgpushvars
#var bUseParentElement $(Typedef.Options.*useParentElement)
#context $(Typedef.DataType)
#var bRValue 1
#var sParamName
#var sParam rtType
#var sdoParam rdoParam
#cginclude "TypeDeserialization.cpp"
#contextend
#cgpopvars
  return rdoParam;
}

#ifeqend // ifeq($(Typedef.Extern),false) // do not serialize extern type
#end
}
